asyncpg/connection.py

```diff
@@ -1222,7 +1222,10 @@ def _on_release(self, stacklevel=1):
     def _drop_local_statement_cache(self):
         self._stmt_cache.clear()
 
-    def _drop_global_statement_cache(self):
+    def _drop_global_statement_cache(self, drop_type_cache=False):
```

Comment:
Make this a keyword-only parameter.

Reply:
Arguably, type cache and statement cache are separate concerns.  I wouldn't lump both of them into a single method.

Reply:
> are separate concerns

Yes, but it has no sense to clear type cache without statement one because statement cache keeps prepared statements which has their own local copy of codecs.

Reply:
It's more nuanced than that.  We are only concerning ourselves with type cache because of composite types, which may change their shape.  We are dealing with two distinct types of failures induced by stale cache: 1) A `RevalidateCachedQuery` failure due to the change of the interpretation of the result type (which may, or may not be a result of an explicit DDL action). 2) Stale composite codec configuration due to an explicit `ALTER TABLE` or `ALTER TYPE`.  IMO, both of these cases deserve separate methods rather than a toggle via a boolean kwarg.

> Yes, but it has no sense to clear type cache without statement one because statement cache keeps prepared statements which have their own local copy of codecs.

Then call `_drop_statement_cache` from `_drop_codec_cache`.